src/main/app/src/app/admin/smart-documents.service.ts

```diff
@@ -82,4 +109,41 @@ export class SmartDocumentsService {
         catchError((err) => this.foutAfhandelingService.foutAfhandelen(err)),
       );
   }
+
+  /**
+   * Flattens a nested RootObject (DocumentsTemplateGroup) into an array of group objects,
+   * omitting nested groups, and preserving templates.
+   * @param {RootObject} obj - The root object to flatten.
+   * @returns {Array<Omit<DocumentsTemplateGroup, "groups">>} - The flattened array of groups with templates, excluding nested groups.
+   */
+  flattenObject(
+    obj: RootObject,
+  ): Array<Omit<DocumentsTemplateGroup, "groups">> {
+    const result: Array<Omit<DocumentsTemplateGroup, "groups">> = [];
+
+    function flattenGroups(group: DocumentsTemplateGroup) {
```

Comment:
Maybe clearer to rename it to `flattenDocumentsTemplateGroup` ? Because that it what it seems to do no?

Reply:
Agreed; changed in next push